meshtastic/stream_interface.py

```
@@ -213,20 +327,29 @@ def __reader(self) -> None:
+            else:
+                disconnect_source = "stream.close_requested"
         except OSError as ex:
             if (
                 not self._wantExit
             ):  # We might intentionally get an exception during shutdown
-                logger.error(
-                    f"Unexpected OSError, terminating meshtastic reader... {ex}"
+                disconnect_source = "stream.os_error"
+                logger.exception(
+                    "Unexpected OSError, terminating meshtastic reader... %s",
+                    ex,
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)